neurolib/models/multimodel/builder/base/backend.py

```diff
@@ -0,0 +1,590 @@
+    def _set_constant_past(self, past_state):
+        self.dde_system.constant_past(past_state)
+        self.dde_system.adjust_diff()
+
+    def _integrate_blindly(self, max_delay):
+        self.dde_system.integrate_blindly(max_delay * 1.5)
+
+    def _check(self):
+        if self.dde_system is not None:
+            self.dde_system.check()
+
+    def run(
+        self, duration, sampling_dt, noise_input, time_spin_up=0.0, **kwargs,
+    ):
+        """
+        Run integration.
+
+        :kwargs:
+            - use_open_mp: OpenMP parallelization - significant overhead, only
+                useful (probably) for very large systems
+            - save_compiled_to: path to which save compiled C code -
+                derivatives and helpers; for large networks this saves the
+                compilation time given multiple runs
+            - load_compiled: whether to try load compiled C code, i.e. do not
+            compile before the run
+        """
+        chunksize = kwargs.pop("chunksize", 1)
+        use_open_mp = kwargs.pop("use_open_mp", False)
+        save_compiled_to = kwargs.pop("save_compiled_to", None)
+        load_compiled = kwargs.pop("load_compiled", False)
+        assert isinstance(noise_input, CubicHermiteSpline)
+
+        if load_compiled:
+            self._init_load_compiled(
+                compiled_dir=save_compiled_to,
+                derivatives=self._derivatives(),
+                helpers=self._sync(),
+                inputs=noise_input,
+                max_delay=self.max_delay,
+                callbacks=self._callbacks(),
+            )
+        else:
+            self._init_and_compile_C(
+                derivatives=self._derivatives(),
+                helpers=self._sync(),
+                inputs=noise_input,
+                max_delay=self.max_delay,
+                callbacks=self._callbacks(),
+                chunksize=chunksize,
+                use_open_mp=use_open_mp,
+            )
+        assert self.dde_system is not None
+        self.dde_system.reset_integrator()
+        logging.info("Setting past of the state vector...")
+        self._set_constant_past(self.initial_state)
+        # integrate
+        times = time_spin_up + np.arange(sampling_dt, duration + sampling_dt, sampling_dt)
+        logging.info(f"Integrating for {times.shape[0]} time steps...")
+        result = np.vstack([self.dde_system.integrate(time) for time in tqdm(times)])
+        if save_compiled_to is not None and not load_compiled:
+            os.makedirs(save_compiled_to, exist_ok=True)
+            logging.info(f"Saving compiled C code to {save_compiled_to}")
+            self.dde_system.save_compiled(
+                os.path.join(save_compiled_to, f"{self.label}.so"), overwrite=True,
+            )
+        return times, result
+
+    def clean(self):
+        """
+        Clean - i.e. remove temp directory from C compilation.
+        """
+        self.dde_system.__del__()
+
+
+class BackendIntegrator:
```

Comment:
This is the main backend class - it is intended as a so-called mixin class, so e.g. `Node` and `Network` inherit from this